omegaconf/dictconfig.py

```diff
@@ -574,6 +574,7 @@ def _set_value(self, value: Any, flags: Optional[Dict[str, bool]] = None) -> Non
             self.__dict__["_content"] = "???"
             self._metadata.object_type = None
         else:
+            previous_state = copy.copy(self.__dict__["_content"])
```

Comment:
1. Add a comment saying this is shallow copy.
2. better variable names are `content_backup` or `prev_content`.